### PR TITLE
Fix loading local development s3proxy files (+ content-security-header changes)

### PR DESCRIPTION
**Pull request by zaccolley**
The main problem this PR is solving is that we can't load our mocked S3 proxy files on the /listen endpoint.

There's various tidying up surrounding this feature in this PR too.

+ Don't test generated .js files
+ Various tidying up in the server file
+ Move Content Security Policy header to seperate file
+ Add development s3proxy CSP changes
+ Don't allow unsafe-inline or unsafe-eval in production (Need to test this)

### CODE REVIEW

File: server/src/csp-header-value.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    'https://*.allizom.org',
+    'data:',
+  ],
+  'media-src': [
+    'data:',
+    'blob:',
+    'https://*.amazonaws.com',
+    'https://*.amazon.com',
+  ],
+  // Note: we allow unsafe-eval locally for certain webpack functionality.
```
Comment (zaccolley) [inline server/src/csp-header-value.ts:23]: move this to the relevant line

File: server/src/csp-header-value.test.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    const CSP_HEADER_VALUE = getCSPHeaderValue();
+
+    expect(CSP_HEADER_VALUE).not.toContain(S3_LOCAL_DEVELOPMENT_ENDPOINT);
+    expect(CSP_HEADER_VALUE).not.toContain('unsafe');
+    expect(CSP_HEADER_VALUE).toBe(
+      `default-src 'none';child-src 'self' blob:;style-src 'self' https://fonts.googleapis.com;img-src 'self' www.google-analytics.com www.gstatic.com https://www.gstatic.com https://*.amazonaws.com https://*.amazon.com https://gravatar.com https://*.mozilla.org https://*.allizom.org data:;media-src data: blob: https://*.amazonaws.com https://*.amazon.com;script-src 'self' 'sha256-fIDn5zeMOTMBReM1WNoqqk2MBYTlHZDfCh+vsl1KomQ=' 'sha256-Hul+6x+TsK84TeEjS1fwBMfUYPvUBBsSivv6wIfKY9s=' https://www.google-analytics.com https://pontoon.mozilla.org https://sentry.prod.mozaws.net;font-src 'self' https://fonts.gstatic.com;connect-src 'self' blob: https://pontoon.mozilla.org/graphql https://*.amazonaws.com https://*.amazon.com https://www.gstatic.com https://www.google-analytics.com https://sentry.prod.mozaws.net https://basket.mozilla.org https://basket-dev.allizom.org https://rs.fullstory.com https://edge.fullstory.com`
```
Comment (zaccolley) [inline server/src/csp-header-value.test.ts:29]: I copied this from production

File: server/src/server.ts [hunk 3]
```diff
@@ -12,47 +16,26 @@ import {
...
-require('source-map-support').install();
 const contributableLocales = require('locales/contributable.json');
 
 const MAINTENANCE_VERSION_KEY = 'maintenance-version';
 const FULL_CLIENT_PATH = path.join(__dirname, '..', '..', 'web');
 const MAINTENANCE_PATH = path.join(__dirname, '..', '..', 'maintenance');
-const RELEASE_VERSION = getConfig().RELEASE_VERSION;
-const ENVIRONMENT = getConfig().ENVIRONMENT;
-const PROD = getConfig().PROD;
+const { RELEASE_VERSION, ENVIRONMENT, SENTRY_DSN, PROD } = getConfig();
```
Comment (mozgzh) [inline server/src/server.ts:34]: I love object destructuring, much cleaner approach :rocket:

**Review by mozgzh — APPROVED** [review]
Changes look good! Considering how important they are, we will need to do thorough testing.

**Pull request merged**